### PR TITLE
BaSP-TG-140: Delete account employee

### PR DESCRIPTION
**Pull request by matescarabino**
- add "isDeleted" field to models
- modify admins controller to new logic
- modify employees controller to new logic
- modify projects controller to new logic

How to test?
- with postman delete a admin, employee or project.
- the getAll method now will no longer show entitys with "isDeleted: true"
- after testing go to mongoDB and revert the changes to the isDeleted fields

### CODE REVIEW

**Review by CristianBucci — APPROVED** [review]
Everything looks right!

File: src/controllers/employee.js [hunk 1]
```diff
@@ -110,9 +115,9 @@ const editEmployee = async (req, res) => {
 
 const deleteEmployee = async (req, res) => {
   try {
-    const employee = await Employee.findById(req.params.id);
-    await firebaseApp.auth().deleteUser(employee.firebaseUid);
-    const result = await Employee.findByIdAndDelete(req.params.id);
+    // const employee = await Employee.findById(req.params.id);
```
Comment (Ansitox) [inline src/controllers/employee.js:118]: Delete comments

File: src/controllers/admins.js [hunk 2]
```diff
@@ -31,11 +31,17 @@ const getAdminById = async (req, res) => {
       throw {
         message: 'Admin not found.', status: 404,
       };
+    } else if (admin.isDeleted === true) {
```
[the remark this answers is not in the document]
Comment (CristianBucci) [reply src/controllers/admins.js:34]: Please check this line again!

**Review by Ansitox — APPROVED** [review]

File: src/controllers/employee.js [hunk 3]
```diff
@@ -31,9 +31,14 @@ const getEmployeeById = async (req, res) => {
       throw {
         message: 'Employee not found.', status: 404,
       };
+    } else if (employee.isDeleted === true) {
```
Comment (CristianBucci) [inline src/controllers/employee.js:34]: Same as above!

File: src/controllers/projects.js [hunk 4]
```diff
@@ -35,11 +35,17 @@ const getProjectById = async (req, res) => {
       throw {
         message: 'Project not found.', status: 404,
       };
+    } else if (project.isDeleted === true) {
```
Comment (CristianBucci) [inline src/controllers/projects.js:38]: Same as above ` (project.isDeleted)`

**Review by CristianBucci — DISMISSED** [review]
Besides those little corrections, everything looks right!

File: src/controllers/admins.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (matescarabino) [reply src/controllers/admins.js:34]: Done

File: src/controllers/employee.js (hunk 1 above)
Comment (matescarabino) [reply src/controllers/employee.js:118]: Done

File: src/controllers/employee.js (hunk 3 above)
Comment (matescarabino) [reply src/controllers/employee.js:34]: Done

File: src/controllers/projects.js (hunk 4 above)
Comment (matescarabino) [reply src/controllers/projects.js:38]: Done

File: src/controllers/admins.js [hunk 5]
```diff
@@ -109,9 +115,9 @@ const editAdmin = async (req, res) => {
 
 const deleteAdmin = async (req, res) => {
   try {
-    const admin = await Admins.findById(req.params.id);
-    await firebaseApp.auth().deleteUser(admin.firebaseUid);
-    const result = await Admins.findByIdAndDelete(req.params.id);
+    // const admin = await Admins.findById(req.params.id);
+    // await firebaseApp.auth().deleteUser(admin.firebaseUid);
```
Comment (CristianBucci) [inline src/controllers/admins.js:119]: Delete these comments!

**Review by NicoSan02 — APPROVED** [review]
🕵️‍♂️ QA FEEDBACK

BaSP-TG-140: Delete account employee

We wanted to tell you that we consider that your PR complies the requirements to be approved and that it can be merged into the master branch 👍

File: src/controllers/admins.js (hunk 5 above)
Comment (matescarabino) [reply src/controllers/admins.js]: Done

**Review by renzopimentel — APPROVED** [review]
Looks good!

**Pull request merged**